### PR TITLE
Fix/aid list bug

### PR DESCRIPTION
**Pull request by mazzeaar**
Fixed a bug where the AidList always returned `total_num_particles` if we called  `getNumParticlesInOctantsParallel` from a rank other than rank 0. I also renamed all _paralell_ i could find to _parallel_....

We now test that this behaviour is correct.

```cpp

std::vector<morton_code> octants_buff(size_buff);
mpi::Status octants_status;
Comm->recv(octants_buff.data(), size_buff, rank, 0, octants_status);

// ... more code

for (size_t i = 0; i < size_buff; ++i) {
    // wrong -> we always checked the octants from rank 0 and not the buff...
    // weights[i] = getNumParticlesInOctant(octants[i]); 

    // right-> we now check the actual octants that we should check
    weights[i] = getNumParticlesInOctant(octants_buff[i]);
}
```

**note**
I also added new comments to the readme

# plz merge directly if review is ok<3

### CODE REVIEW

**Comment by mazzeaar** [comment]
this should finally close #29

File: src/OrthoTree/helpers/AidList.hpp [hunk 1]
```diff
@@ -199,36 +199,48 @@ namespace ippl {
...
+
+                // no need to shrink, we only send what we need
+                if (weights.size() < size_buff) {
+                    weights.resize(size_buff);
+                }
 
-                for (auto octant : octants_buff) {
-                    weights.push_back(getNumParticlesInOctant(octant));
+                for (size_t i = 0; i < size_buff; ++i) {
+                    weights[i] = getNumParticlesInOctant(octants_buff[i]);
```
Comment (mazzeaar) [inline src/OrthoTree/helpers/AidList.hpp:223]: the bug i fixed is on this line (223), the rest of this files changes is formatting and comments

```cpp
weights[i] = getNumParticlesInOctant(octants_buff[i]);
```

File: unit_tests/OrthoTree/helpers/aid_list.cpp [hunk 2]
```diff
@@ -231,74 +232,84 @@ TEST(AidListTest, NumParticlesInOctantTest) {
...
+
+                                            // middle-middle octant
+                                            {.octant                    = 0b111,
+                                             .expected_lower_bound      = 300,
+                                             .expected_upper_bound_excl = 400,
+                                             .expected_upper_bound_incl = 399,
+                                             .expected_total_particles  = n_particles_per_proc}};
+
+    // #### TEST HELPERS DONE ####
+
```
Comment (mazzeaar) [inline unit_tests/OrthoTree/helpers/aid_list.cpp:294]: i moved this outside of the `rank == 0` bounds below, as the other ranks now also need this data

File: unit_tests/OrthoTree/helpers/aid_list.cpp [hunk 3]
```diff
@@ -231,74 +232,84 @@ TEST(AidListTest, NumParticlesInOctantTest) {
...
             run_test(test_data);
         }
     }
+
+    for (auto test_data : tests_to_run) {
+        std::vector<morton_code> octants = {test_data.octant};
+
+        // this has always size one, so we can just take the first element in the returning vector
+        const size_t calc_result = aid_list.getNumParticlesInOctantsParallel(octants).front();
+        EXPECT_EQ(test_data.expected_total_particles, calc_result)
```
Comment (mazzeaar) [inline unit_tests/OrthoTree/helpers/aid_list.cpp:309]: we now additionally check that each rank gets the correct number of particles in its octant if we call it in parallel

**Review by casparschucan — APPROVED** [review]
Nice spot

**Pull request merged**